packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -231,6 +239,27 @@ abstract class BindingBase {
     assert(!locked);
   }
 
+  /// Cause the entire application to stop drawing, e.g. in preparation for hot
+  /// reload.
+  ///
+  /// This is used by development tools when the application code is being
+  /// changed, to ensure that animations will not continue to allow builds to
+  /// fire before `ext.flutter.reassemble` is sent. It can be triggered
+  /// manually by sending the `ext.flutter.waitForReassemble` service extension
+  /// signal.
+  ///
+  /// This method must never be used in production code, as it will prevent all
```

Comment:
If this should never be called, should the body be wrapped in an assert?

Reply:
That sounds reasonable to me.  It does make me wonder why `reassembleApplication` is not wrapped in an assert, but that shouldn't stop us from ding it here.

Reply:
Done